src/hal/sensor/sensor.cpp

```diff
@@ -0,0 +1,80 @@
+
+#include "../../hal/hal.h"
+#include "../../hal/sensor/sensor.h"
+
+// Sensor pin defines
+#define FLOW_SENSE_PIN     A0
+#define PRESSURE_SENSE_PIN A1
+
+
+// The following function initializes Timer 5 to do pressure and flow sensor data collection
+int16_t sensorHalInit(void)
+{
+  // Sensor timer
+  TCCR5A=0; // set timer control registers
+  TCCR5B=0x0A;  // set timer to clear timer on compare (CTC), and prescaler to divide by 8
+  TCNT5=0;  // initialize counter to 0
+  OCR5A= 1000;  // Set interrupt to 2khz
+  TIMSK5 |= 0x02; // enable timer compare interrupt
+
+  return HAL_OK;
+}
+
+
+//--------------------------------------------------------------------------------------------------
+int16_t pSum; // pressure data accumulator
+// This routine reads the pressure value as a voltage off the analog pin and inserts it into the filter
+void pressureSensorHalFetch(){
+  int16_t pIn;  // pressure value from sensor.
+  pIn=analogRead(PRESSURE_SENSE_PIN);
+  pSum = pSum-(pSum>>4)+pIn; // filter
+}
+
+// This routine returns the pressure value out of the filter
+int16_t pressureSensorHalGetValue(int16_t *value){
+  int32_t temp = pSum>>4;
+  int32_t pascals = ((temp * 217L) - 110995L)>>2;	// convert to Pascals
+  int32_t u100umH2O = (pascals * 4177L)>>12;		// convert Pascals to 0.1mmH2O
+  *value = (int16_t)u100umH2O;						// return as 16 bit signed int
+  return HAL_OK;
+}
+
+
+//--------------------------------------------------------------------------------------------------
+int16_t fSum; // flow sensor data accumulator
+// This routine reads the flow sensor value as a voltage off the analog pin and inserts it into the filter
+void airflowSensorHalFetch(){
+  int16_t fIn;  // pressure value from sensor.
+  fIn=analogRead(FLOW_SENSE_PIN);
+  fSum = fSum-(fSum>>3)+fIn; // filter
+}
+// This routine returns the flow sensor value out of the filter
+int16_t airflowSensorHalGetValue(int16_t *value){
+  *value = fSum>>3;
```

Comment:
See above about atomic access ,`fSum` needs it too

Reply:
Done